index.ts

```diff
@@ -0,0 +1,154 @@
+    aws.iam.Principals.LambdaPrincipal
+  ),
+  managedPolicyArns: [
+    aws.iam.ManagedPolicy.AWSLambdaBasicExecutionRole,
+    aws.iam.ManagedPolicy.AWSLambdaVPCAccessExecutionRole,
+  ],
+  tags: tags,
+});
+
+const infoHandler = new aws.lambda.Function("info-handler", {
+  role: lambdaRole.arn,
+  runtime: aws.lambda.Runtime.Python3d9,
+  code: new pulumi.asset.FileArchive("./lambda/main.py.zip"),
+  handler: "main.info_handler", // fileName.methodName
+  vpcConfig: {
+    subnetIds: [privateSubnet.id],
+    securityGroupIds: [vpc.defaultSecurityGroupId],
+  },
+  tags: tags,
+});
+
+// API Gateway
+const securityGroup = new aws.ec2.SecurityGroup("api-gateway-security-group", {
+  vpcId: vpc.id,
+  ingress: [
+    {
+      description: "HTTP inbound",
+      fromPort: 80,
+      toPort: 80,
+      protocol: "tcp",
+      cidrBlocks: ["0.0.0.0/0"],
+    },
+  ],
+  tags: tags,
+});
+
+const vpcLink = new aws.apigatewayv2.VpcLink("andy-vpc-link", {
+  subnetIds: [privateSubnet.id],
+  securityGroupIds: [securityGroup.id],
+  tags: tags,
+});
+
+const api = new aws.apigatewayv2.Api("andy-api-gateway", {
+  protocolType: "HTTP",
+  tags: tags,
+});
+
+//// https://github.com/pulumi/examples/blob/master/aws-ts-apigatewayv2-http-api/index.ts
+const withApi = new aws.lambda.Permission(
+  "api-lambda-permission",
+  {
+    action: "lambda:InvokeFunction",
+    principal: "apigateway.amazonaws.com",
+    function: infoHandler,
+    sourceArn: pulumi.interpolate`${api.executionArn}/*/*`,
+  },
+  { dependsOn: [api, infoHandler] }
+);
+
+const integration = new aws.apigatewayv2.Integration("andy-integration", {
+  apiId: api.id,
+  integrationType: "AWS_PROXY",
+  integrationUri: infoHandler.arn,
+  integrationMethod: "GET",
+});
+
+const route = new aws.apigatewayv2.Route("get-info-route", {
+  apiId: api.id,
+  routeKey: "GET /info",
+  target: integration.id.apply((id) => "integrations/" + id),
+});
+
+const stage = new aws.apigatewayv2.Stage("api-stage", {
+  apiId: api.id,
+  name: stack,
+  autoDeploy: true,
+});
+
+// Create Internet Gateway, NAT Gateway and edit route tables
+const igw = new aws.ec2.InternetGateway("igw", {
+  vpcId: vpc.id,
+  tags: tags,
+});
+
+const natEip = new aws.ec2.Eip("nat-eip", {
+  vpc: true,
+  tags: tags,
+});
+
+const natgw = new aws.ec2.NatGateway("natgw", {
+  subnetId: publicSubnet.id,
+  allocationId: natEip.id,
+  tags: tags,
+});
+
+const privateRouteTable = new aws.ec2.RouteTable("private-route-table", {
+  vpcId: vpc.id,
+  routes: [{ cidrBlock: "0.0.0.0/0", natGatewayId: natgw.id }],
+  tags: tags,
+});
+
+const privateAssociation = new aws.ec2.RouteTableAssociation(
+  "private-subnet-association",
+  {
+    routeTableId: privateRouteTable.id,
+    subnetId: privateSubnet.id,
+  }
+);
+
+const publicRouteTable = new aws.ec2.RouteTable("public-route-table", {
+  vpcId: vpc.id,
+  routes: [{ cidrBlock: "0.0.0.0/0", gatewayId: igw.id }],
+  tags: tags,
```

Comment:
Nice things pulumi has to automate tags creation: https://www.pulumi.com/blog/automatically-enforcing-aws-resource-tagging-policies/#automatically-applying-tags